src/DataGrid.tsx

```diff
@@ -426,10 +446,29 @@ function DataGrid<R, K extends keyof R, SR>({
     const { key } = event;
     const column = columns[selectedPosition.idx];
     const row = rows[selectedPosition.rowIdx];
-    const canOpenEditor = selectedPosition.mode === 'SELECT' && isCellEditable(selectedPosition);
-    const isActivatedByUser = (column.unsafe_onCellInput ?? legacyCellInput)(event, row) === true;
+    const isUsingEditor2 = column.editor2 !== undefined;
+
+    if (selectedPosition.mode === 'EDIT') {
+      if (isUsingEditor2 && (key === 'Enter' || key === 'F2')) {
+        handleCommit2();
+      }
+      return;
+    }
 
-    if (canOpenEditor && (key === 'Enter' || isActivatedByUser)) {
+    const isActivatedByUser = isUsingEditor2
+      ? defaultCellInput(event) && column.editor2Props?.onCellInput?.(event) !== false
```

Comment:
I think we should open the editor on `Enter`, `F2`, or on any other alpha numeric key. If the users need to restrict it then they can use `onCellInput` . This is also the recommended wau
https://www.w3.org/TR/wai-aria-practices/#gridNav_inside